src/brsTypes/components/RoSGNode.ts

```diff
@@ -317,6 +317,45 @@ export class RoSGNode extends BrsComponent implements BrsValue, BrsIterable {
         return false;
     }
 
+    /**
+     * Calls the function specified on this node.
+     */
+    private callfunc = new Callable("callfunc", {
+        signature: {
+            args: [
+                new StdlibArgument("functionname", ValueKind.String),
+                new StdlibArgument("functionargs", ValueKind.Dynamic, Uninitialized.Instance),
+            ],
+            returns: ValueKind.Dynamic,
+        },
+        impl: (interpreter: Interpreter, functionname: BrsString, functionargs: BrsType) => {
+            // We need to search the callee's environment for this function rather than the caller's.
+            let componentDef = interpreter.environment.nodeDefMap.get(this.nodeSubtype);
+            if (componentDef) {
+                return interpreter.inSubEnv(subInterpreter => {
+                    let functionToCall = subInterpreter.getCallableFunction(functionname.value);
+
+                    // Args for callFunc are optional, they're based on the function we're calling,
+                    // and we need to match the signature of that function, so we can't always forward functionargs.
+                    //
+                    // You _can_ pass "invalid" as an argument to a function, so we'll use Uninitialized
+                    // as a default value here to differentiate.
+                    if (functionargs == Uninitialized.Instance) {
```

Comment:
The comment pretty much sums it up, but let me know if it doesn't make sense or there's a better solution.